PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalInternalClient.kt

```diff
@@ -131,6 +137,102 @@ internal class PayPalInternalClient(
     }
 
+    @ExperimentalBetaApi
+    fun sendVaultEditRequest(
+        context: Context,
+        request: PayPalVaultEditRequest,
+        callback: PayPalInternalClientEditCallback
+    ) {
+        getClientMetadataId(
+            context
+        ) { clientMetadataId ->
+            if (clientMetadataId == null) {
+                callback.onPayPalVaultEditResult(null, BraintreeException("An unexpected error occurred"))
+            } else {
+
+                val riskCorrelationId =
+                    (request as? PayPalVaultErrorHandlingEditRequest)
+                        ?.riskCorrelationId ?: clientMetadataId
+
+                sendVaultEditRequestWithRiskCorrelationId(
+                    request,
+                    riskCorrelationId,
+                    callback
+                )
+            }
+        }
+    }
+
+    @ExperimentalBetaApi
+    private fun sendVaultEditRequestWithRiskCorrelationId(
+        payPalVaultEditRequest: PayPalVaultEditRequest,
+        riskCorrelationId: String,
+        callback: PayPalInternalClientEditCallback
+    ) {
+        val params = parameters(payPalVaultEditRequest.editPayPalVaultId).toMutableMap()
+
+        params["risk_correlation_id"] = riskCorrelationId
+
+        val jsonObject = JSONObject(params.toMap())
+
+        braintreeClient.sendPOST(
+            payPalVaultEditRequest.hermesPath,
+            jsonObject.toString()
+        ) { response, error ->
+            if (error != null) {
+                callback.onPayPalVaultEditResult(null, error)
+            } else {
+                try {
+                    val responseBody = JSONObject(response)
+                    val agreementSetup = responseBody.getJSONObject("agreementSetup")
+
+                    val params = PayPalVaultEditAuthRequestParams(
+                        riskCorrelationId,
+                        null,
+                        agreementSetup.getString("approvalUrl"),
+                        successUrl
+                    )
+
+                    callback.onPayPalVaultEditResult(params, null)
+                } catch (jsonException: JSONException) {
+                    callback.onPayPalVaultEditResult(null, jsonException)
+                }
+            }
+        }
+    }
+
+    // TODO: improve method name
+    fun parameters(editPayPalVaultId: String): Map<String, Any> {
+        val parameters = mutableMapOf<String, Any>()
+
+        parameters["edit_paypal_vault_id"] = editPayPalVaultId
+        parameters["return_url"] = successUrl
+        parameters["cancel_url"] = cancelUrl
+
+        return parameters
+    }
+
+    private fun getClientMetadataId(
+        context: Context,
+        callback: (String?) -> Unit
+    ) {
+        braintreeClient.getConfiguration { configuration, error ->
+
+            if (error != null) {
+                callback(error("No Client Metadata Id"))
+            } else {
+
+                // TODO: what to do with hasUserLocationConsent
```

Comment:
Should we address this TODO in this PR?

Reply:
I talked to @sshropshire and we can leave as is. I'll remove the TODO.